benchmarks/common/benchmark_runner.hpp

```diff
@@ -293,86 +303,38 @@ struct BenchmarkRunner {
-
-    void initialize(const ProblemShapeType& problem_size) override {
-      Base::initialize(problem_size);
-    }
+      std::stringstream full_test_name;
+      full_test_name << test_name << "/";
+      std::string test_name_suffix = std::to_string(options.m) + "x" + 
+                                     std::to_string(options.n) + "x" + 
+                                     std::to_string(options.k) + "x" + 
+                                     std::to_string(options.l);
+      full_test_name << test_name_suffix;
+      benchmark::RegisterBenchmark(full_test_name.str().c_str(), run_benchmark, options, gemm_op)
+      ->UseManualTime();
+      benchmark::RunSpecifiedBenchmarks();
+  }
+
+  ~BenchmarkRunner() {
+    benchmark::Shutdown();
+  }
     
-    void run(const Options& options, const cutlass::KernelHardwareInfo& hw_info) override {
-      ProblemShapeType problem_size = ProblemShapeType{options.m, options.n, options.k, options.l};
-
-      initialize(problem_size);
-
-      typename Gemm::GemmKernel::Arguments arguments{
-              cutlass::gemm::GemmUniversalMode::kGemm,
-              problem_size,
-              {Base::block_A.get(), Base::stride_A, Base::block_B.get(), Base::stride_B},
-              {
-                {options.alpha, options.beta},
-                Base::block_C.get(), Base::stride_C, Base::block_D.get(), Base::stride_D
-              },
-              hw_info
-      };
-
-      Gemm gemm_op;
-
-      size_t workspace_size = Gemm::get_workspace_size(arguments);
-      cutlass::device_memory::allocation<uint8_t> workspace(workspace_size);
-
-      gemm_op.can_implement(arguments);
-
-      gemm_op.initialize(arguments, workspace.get());
-
-      // Run the GEMM
-      gemm_op.run();
-
-#if defined(CUTLASS_ENABLE_SYCL)
-      syclcompat::wait();
-#else
-      cudaDeviceSynchronize();
-#endif
-
-      // Verify that the result is correct
-      bool passed = Base::verify(problem_size, options.alpha, options.beta);
-      std::cout << "Disposition: " << (passed ? "Passed" : "Failed") << std::endl;
-
-      if (passed && options.iterations > 0) {
+  private:
+    static void run_benchmark(benchmark::State& state, const Options& options,  Gemm gemm_op) {
+      for(auto _ : state) {
         GPU_Clock timer;
         timer.start();
-        for (int i = 0; i < options.iterations; ++i) {
-          gemm_op.run();
-        }
-
-        float cute_time = timer.seconds() / options.iterations;
-        double tflops = (2.0 * options.m * options.n * options.k * options.l) * 1e-12;
-        std::cout << "Problem Size: " << options.m << 'x' << options.n << 'x' << options.k << 'x' << options.l << std::endl;
-        printf("Cutlass GEMM Performance:     [%4.3f]TFlop/s  (%6.4f)ms\n", tflops / cute_time, cute_time*1000);
+        gemm_op.run(); 
+        float cute_time = timer.seconds();
+        state.counters["TFlops"] = ((2.0 * options.m * options.n * options.k * options.l) * 1e-12) / cute_time;
```

Comment:
This will only contain the TFLOPs calculation from the last iteration. You can refer to the portBlas benchmarks for adding counters (they can be added as a separate ticket/PR).

- https://github.com/codeplaysoftware/portBLAS/blob/fbce030a68d464021503c35cd04c4c1be308faba/benchmark/portblas/blas3/gemm.cpp#L125

- https://github.com/codeplaysoftware/portBLAS/blob/fbce030a68d464021503c35cd04c4c1be308faba/common/include/common/common_utils.hpp#L1913

- https://github.com/codeplaysoftware/portBLAS/blob/fbce030a68d464021503c35cd04c4c1be308faba/common/include/common/blas3_state_counters.hpp#L67

```suggestion
```

Reply:
Oh, I was not aware of that. Turns out in that case, we have been not measuring avg time in portFFT too, which I had been using as a [reference](https://github.com/codeplaysoftware/portFFT/blob/653fde89ca16f3b08a925c8a06d1f1a391e90f21/test/bench/portfft/launch_bench.hpp#L138) 

BTW, would the latest [commit](https://github.com/codeplaysoftware/cutlass-fork/pull/116/commits/4e7ad1db8223eda26fa3a52afbf402362de1fb94) fix it ?